### PR TITLE
[Enhancement]Support delete txn_log when publish batch succeed

### PR DESCRIPTION
**Pull request by smartlxh**
Why I'm doing:
Pr #28789 will not delete remove txn_log when enable publish batch, and let the vacuum do the work. 

What I'm doing:
The pr will delete the txn_log when publish succeed by send a rpc to Be to alleviate the pressure of vacuum,
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/storage/lake/vacuum.cpp [hunk 1]
```diff
@@ -594,4 +594,22 @@ void delete_tablets(TabletManager* tablet_mgr, const DeleteTabletRequest& reques
     st.to_protobuf(response->mutable_status());
 }
 
+void delete_txn_log(TabletManager* tablet_mgr, const DeleteTxnLogRequest& request, DeleteTxnLogResponse* response) {
+    DCHECK(tablet_mgr != nullptr);
+    DCHECK(request.tablet_ids_size() > 0);
+    DCHECK(response != nullptr);
+
+    std::vector<std::string> files_to_delete;
```
Comment (sduzh) [inline be/src/storage/lake/vacuum.cpp:602]: files_to_delete.reserve(request.tablet_ids_size() * request.txn_ids_size());

File: gensrc/proto/lake_service.proto [hunk 2]
```diff
@@ -62,6 +62,15 @@ message DeleteTabletResponse {
     optional StatusPB status = 2;
 }
 
+message DeleteTxnLogRequest {
+    repeated int64 tablet_ids = 1;
+    repeated int64 txn_ids = 2;
+}
+
+message DeleteTxnLogResponse {
+    optional StatusPB status = 2;
```
Comment (sduzh) [inline gensrc/proto/lake_service.proto:71]:
```suggestion
    optional StatusPB status = 1;
```

File: fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java [hunk 3]
```diff
@@ -563,6 +595,46 @@ public boolean publishPartitionBatch(Database db, long tableId, long partitionId
...
+                        Future<DeleteTxnLogResponse> future = lakeService.deleteTxnLog(request);
+                        responseList.add(future);
+                    }
+
+                    for (Future<DeleteTxnLogResponse> deleteTxnLogResponseFuture : responseList) {
+                        DeleteTxnLogResponse response = deleteTxnLogResponseFuture.get();
+                        if (response.status.errorMsgs != null && !response.status.errorMsgs.isEmpty()) {
+                            LOG.warn("delete txn log error: " + response.status.errorMsgs);
+                        }
+                    }
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java:630]: Can we just ignore the response?

File: be/src/storage/lake/vacuum.cpp (hunk 1 above)
Comment (smartlxh) [reply be/src/storage/lake/vacuum.cpp:602]: done

File: fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java [hunk 4]
```diff
@@ -563,6 +595,45 @@ public boolean publishPartitionBatch(Database db, long tableId, long partitionId
         return true;
     }
 
+    private void submitDeleteTxnLogJob(TransactionStateBatch txnStateBatch, Map<Long, List<Long>> dirtyPartitions) {
+        getDeleteTxnLogExecutor().submit(() -> {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java:599]: `submit` may throw [RejectedExecutionException](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/RejectedExecutionException.html) if the task cannot be scheduled for execution.

**Review by sduzh — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**